File: Wiki/C/code/C_Primer_Plus/05-Operators_Expressions_and_Statements/07.c
// Purpose: precedence test
// Reference: page 156 (paper) / 185 (ebook)

#include <stdio.h>

    int
main(void)
{
    int top, score;

    top = score = -(2 + 5) * 6 + (4 + 3 * (2 + 3));
    printf("top = %d, score = %d\n", top, score);

    return 0;
}

File: Wiki/lua/code/Programming_in_Lua_4th_edition/ch01_Getting_Started/exercises/07.lua
-- Purpose: Consider the following expression:
--
--     (x and y and (not z)) or ((not y) and x)
--
-- Are  the  parentheses necessary?   Would  you  recommend  their use  in  that
-- expression?

-- Reference: page 11 (paper) / 25 (ebook)


-- No, the parens  are not required, because the operations  would be grouped in
-- the  same way  without.  Indeed,  according  to their  precedence level,  the
-- operators are ranked in that order: `not`, `and`, `or`.
-- Still, it's better  to use parens, because it removes  any confusion that the
-- reader might have with regard to how the operations are grouped.
